manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/provider/CustomSnippetsFieldDataProvider.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.generator.services.provider;
+
+import io.hyscale.commons.framework.patch.FieldMetaData;
+import io.hyscale.commons.framework.patch.FieldMetaDataProvider;
+import io.hyscale.servicespec.commons.builder.MapFieldDataProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Provides field data map which contains primary key field for an entity
+ * Used to distinguish between object in operations such as merge for
+ * K8s Patches.
+ *
+ * @author Nishanth Panthangi
+ *
+ */
+public class CustomSnippetsFieldDataProvider implements FieldMetaDataProvider {
+    private static final Logger logger = LoggerFactory.getLogger(MapFieldDataProvider.class);
```

Comment:
Class in logger should be CustomSnippetsFieldDataProvider

Reply:
LOL forgot